File: 1608. Special Array With X Elements Greater Than or Equal X/solution.cpp
#include<iostream>
#include<vector>
using namespace std;


class Solution {
public:
    int specialArray(vector<int>& nums) {
        int n = nums.size();
        int lo = 0, hi = n;
        while(lo <= hi){
            int mid = (lo+hi)>>1;
            int ct = count(nums, mid);
            if( ct == mid)
                return mid;
            else if(ct < mid)
                hi = mid - 1;
            else
                lo = mid + 1;
        }
        return -1;
    }
    
    private: 
    int count(vector<int> &A, int k) {
        int ct = 0;
        for(auto x: A)
            if(x >= k)
                ct++;
        return ct;
    }
};

File: 1608. Special Array With X Elements Greater Than or Equal X/solution.py
from bisect import bisect_left
class Solution:
    def specialArray(self, nums: List[int]) -> int:
        nums.sort()
        n=len(nums)
        final=-1
        for x in range(1,n+1):
            ind=bisect_left(nums,x)
            greater = n-ind
            if x==greater:
                final=x
        return final
            